### PR TITLE
Add IntelliTect analyzers and resolve errors

### LINKED ISSUE #68: Turn on code analysis for Selenate and TestFramework
I forgot when the projects were started. :(

### PR DESCRIPTION
**Pull request by PandaMagnus**
Halfway addresses #68 (a second PR will add analyzers for TestFramework).

### CODE REVIEW

File: IntelliTect.TestTools.Selenate/GoogleSearch/GoogleHarness.cs [hunk 1]
```diff
@@ -13,7 +14,7 @@ public GoogleHarness(GoogleBrowser browser)
             Browser = browser;
         }
 
-        public string URL => "https://www.google.com";
+        public static Uri URL => new Uri("https://www.google.com");
```
Comment (Keboo) [inline IntelliTect.TestTools.Selenate/GoogleSearch/GoogleHarness.cs:17]: Given that this isn't changing you can make it a readonly property.

File: IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs [hunk 2]
```diff
@@ -145,22 +146,33 @@ public void TakeScreenshot()
...
             }
 
             if (!file.Directory.Exists)
             {
-                Console.WriteLine($"Skipping TakeScreenshot. Path does not exist: {file.FullName}");
+                Debug.WriteLine($"Skipping TakeScreenshot. Path does not exist: {file.FullName}");
                 return;
             }
 
+            if(file.Exists)
```
Comment (Keboo) [inline IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs:167]: `file.Delete()` does nothing if the file doesn't exist, so you don't need this check here.

File: IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementHandler.cs [hunk 3]
```diff
@@ -117,6 +117,6 @@ public bool WaitForEnabledState(IWebElement element, int secondsToTry = 5)
             }            
         }
 
-        private IWebDriver _Driver { get; set; }
+        private IWebDriver Driver { get; set; }
```
Comment (Keboo) [inline IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementHandler.cs:120]: It appears this is only set in the ctor and could be made readonly.

File: IntelliTect.TestTools.Selenate/GoogleSearch/GoogleHarness.cs (hunk 1 above)
Comment (PandaMagnus) [reply IntelliTect.TestTools.Selenate/GoogleSearch/GoogleHarness.cs]: Changed to a... auto-property? Can't remember the name. Anyway, it's now URL { get;} = new...

File: IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementHandler.cs (hunk 3 above)
Comment (PandaMagnus) [reply IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ElementHandler.cs]: Removed the setter.

File: IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs (hunk 2 above)
Comment (PandaMagnus) [reply IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/Browser.cs]: Removed the check.

**Comment by PandaMagnus** [comment]
@Keboo  Should we push out a release version of the analyzers now? The build is failing on not finding the pre-release package.

**Pull request merged**